packages/vuetify/src/components/transitions/createTransition.ts

```diff
@@ -45,59 +26,49 @@ export function createSimpleTransition (
       },
     },
 
-    render (h, context): VNode {
-      const tag = `transition${context.props.group ? '-group' : ''}`
-      const data: VNodeData = {
-        props: {
+    setup (props, { slots }) {
+      return () => {
+        const tag = props.group ? TransitionGroup : Transition
+
+        return h(tag as FunctionalComponent, {
           name,
-          mode: context.props.mode,
-        },
-        on: {
-          beforeEnter (el: HTMLElement) {
-            el.style.transformOrigin = context.props.origin
-            el.style.webkitTransformOrigin = context.props.origin
+          mode: props.mode,
+          onBeforeEnter (el: HTMLElement) {
+            el.style.transformOrigin = props.origin
           },
-        },
-      }
-
-      if (context.props.leaveAbsolute) {
-        data.on!.leave = mergeTransitions(data.on!.leave, (el: HTMLElement) => (el.style.position = 'absolute'))
-      }
-      if (context.props.hideOnLeave) {
-        data.on!.leave = mergeTransitions(data.on!.leave, (el: HTMLElement) => (el.style.display = 'none'))
+          onLeave (el: HTMLElement) {
+            if (props.leaveAbsolute) el.style.position = 'absolute'
+            if (props.hideOnLeave) el.style.display = 'none'
+          },
+        }, slots.default)
       }
-
-      return h(tag, mergeData(context.data, data), context.children)
     },
-  }
+  })
 }
 
 export function createJavascriptTransition (
   name: string,
   functions: Record<string, any>,
   mode = 'in-out'
-): FunctionalComponentOptions {
-  return {
+) {
+  return defineComponent({
     name,
 
-    functional: true,
-
     props: {
       mode: {
         type: String,
         default: mode,
       },
     },
 
-    render (h, context): VNode {
-      return h(
-        'transition',
-        mergeData(context.data, {
-          props: { name },
-          on: functions,
-        }),
-        context.children
-      )
+    setup (props, { slots }) {
+      return () => {
+        return h(Transition as any, {
+          name,
+          // mode: props.mode, TODO: vuejs/vue-next#3104
+          ...functions,
+        }, () => slots.default?.())
```

Comment:
I assume this is intended. Can you explain what we have to return a function that returns the default slot instead of just returning the default slot.